File: original/src/test/java/com/realworld/project/application/user/api/CredentialApiTest.java
package com.realworld.project.application.user.api;

import com.realworld.project.application.user.api.dto.RequestLoginUser;
import com.realworld.project.application.user.api.dto.RequestRegisterUser;
import com.realworld.project.fixture.ApiTest;
import io.restassured.http.ContentType;
import io.restassured.response.Response;
import org.hamcrest.Matchers;
import org.junit.jupiter.api.*;

import static com.realworld.project.fixture.UserFixture.*;
import static io.restassured.RestAssured.given;
import static org.hamcrest.Matchers.is;

class CredentialApiTest extends ApiTest {

    @DisplayName("회원등록")
    @Test
    void test_case1() {
        registrationApi(username, email, password);

        response
            .then()
            .statusCode(200)
            .body("user.email", is(email))
            .body("user.username", is(username));
    }

    @DisplayName("회원등록후 로그인")
    @Test
    void test_case2() {
        registrationApi(username, email, password);
        loginApi(email, password);

        response
            .then()
            .statusCode(200)
            .body("user.email", is(email))
            .body("user.username", is(username))
            .body("user.token", Matchers.notNullValue());
    }

    public static void loginApi(String email, String password) {
        response = given()
                .contentType(ContentType.JSON)
                .body(new RequestLoginUser(email, password))
                .when()
                .post("/api/users/login");

        token = response.jsonPath().get("user.token");
    }

    public static void registrationApi(String username, String email, String password) {
        response =  given()
                .contentType(ContentType.JSON)
                .body(new RequestRegisterUser(username, email, password))
                .when()
                .post("/api/users");
    }
}

File: kotlin/src/test/kotlin/com/example/realworld/application/user/api/CredentialApiTest.kt
package com.example.realworld.application.user.api

import com.example.realworld.application.user.api.dto.RequestRegisterUser
import com.example.realworld.fixture.ApiTest
import io.restassured.RestAssured
import io.restassured.http.ContentType
import io.restassured.module.kotlin.extensions.Extract
import io.restassured.module.kotlin.extensions.Given
import io.restassured.module.kotlin.extensions.Then
import io.restassured.module.kotlin.extensions.When
import org.hamcrest.CoreMatchers.equalTo
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class CredentialApiTest : ApiTest() {

    @Test
    fun `회원 추가 dto 테스트`() {
        Given {
            contentType(ContentType.JSON)
            body(RequestRegisterUser(username = "username", email = "email", password = "password"))
        } When {
            post("/api/users")
        } Then {
            statusCode(200)
            body("user.email", equalTo("email"))
        } Extract {
            println(body().jsonPath().prettify())
        }
    }
}
